core/src/main/java/io/questdb/client/Sender.java

```diff
@@ -276,29 +335,30 @@ enum TlsValidationMode {
      * }</pre>
      */
     final class LineSenderBuilder {
+        private static final int AUTO_FLUSH_DISABLED = 0;
+        private static final int AUTO_FLUSH_ROWS_NOT_SET = -1;
         private static final byte BUFFER_CAPACITY_NOT_SET = 0;
+        private static final int DEFAULT_AUTO_FLUSH_ROWS = 600;
         private static final int DEFAULT_BUFFER_CAPACITY = 64 * 1024;
         private static final int DEFAULT_HTTP_PORT = 9000;
         private static final int DEFAULT_HTTP_TIMEOUT = 30_000;
-        private static final int DEFAULT_MAXIMUM_BUFFER_CAPACITY = Integer.MAX_VALUE;
-        private static final int DEFAULT_MAX_PENDING_ROWS = 10_000;
+        private static final int DEFAULT_MAXIMUM_BUFFER_CAPACITY = 100 * 1024 * 1024;
         private static final long DEFAULT_MAX_RETRY_NANOS = TimeUnit.SECONDS.toNanos(10); // keep sync with the contract of the configuration method
         private static final int DEFAULT_TCP_PORT = 9009;
         private static final int HTTP_TIMEOUT_NOT_SET = -1;
```

Comment:
Probably worth documenting these magic values with a global.
I.e. why `-1`? What other values are valid here?

```
/** description of the set/unset workflow here */
private static final int UNSET_TOKEN;
```

Reply:
good one. I am actually not sure why I use a different constant for each PROPERTY_NOT_SET. let me try with just a single one.

Reply:
Oh I see! I didn't realise these were constants. I think I was mis-reading the code.
I thought these were state variables living alongside the field to track if the field was already set. You've just intrusively picked a value to represent "not set" which makes more sense.